### PR TITLE
Refactor 'generic' Bash libraries

### PR DESCRIPTION
**Pull request by juan131**
Refactor existing 'generic' libraries based on Bash best practise guidelines.

### CODE REVIEW

File: stretch/rootfs/libcomponent.sh [hunk 1]
```diff
@@ -1,14 +1,38 @@
...
+set -o pipefail
+
+# Constants
+CACHE_ROOT="/tmp/bitnami/pkg/cache"
+DOWNLOAD_URL="https://downloads.bitnami.com/files/stacksmith"
+
+# Functions
+
+########################
+# Prints the welcome page for a Bitnami Docker image
```
Comment (javsalgar) [inline stretch/rootfs/libcomponent.sh]: Decide between third person and imperative. I see differences in different functions

File: stretch/rootfs/libcomponent.sh [hunk 2]
```diff
@@ -1,14 +1,38 @@
...
+# Globals:
+#   OS_NAME
+#   OS_ARCH
+#   OS_FLAVOUR
```
Comment (javsalgar) [inline stretch/rootfs/libcomponent.sh]: If it is not too much effort, change it to OS_FLAVOR (AmE)

File: stretch/rootfs/libbitnami.sh [hunk 3]
```diff
@@ -1,10 +1,26 @@
...
+# Globals:
+#   DISABLE_WELCOME_MESSAGE
+#   BITNAMI_APP_NAME
+# Arguments:
+#   None
+# Returns:
+#   None
+#########################
 print_welcome_page() {
     if [ -z "$DISABLE_WELCOME_MESSAGE" ]; then
```
Comment (javsalgar) [inline stretch/rootfs/libbitnami.sh]: I see that in the rest you use `[[` instead of `[`

File: stretch/rootfs/libcomponent.sh (hunk 2 above)
Comment (juan131) [reply stretch/rootfs/libcomponent.sh]: God saves the queen! I'll keep it

File: stretch/rootfs/libcomponent.sh (hunk 1 above)
Comment (juan131) [reply stretch/rootfs/libcomponent.sh]: 👍

File: stretch/rootfs/libbitnami.sh (hunk 3 above)
Comment (juan131) [reply stretch/rootfs/libbitnami.sh]: 👍

**Pull request merged**